Azure/sap-hana
# Address new requirements on deployer

## Problem
1. Need to add additional folder structure on deployer.
1. CIDR, region and landscape need to be default in json but not tf files.
1. Need required folder structure on deployer.
1. Default SSH key pair is pointing to `~/.ssh/id_rsa*`, which is not desired.
1. local deployer state should use default naming: `terraform.tfstate`
1. Required to upload `tfstate/json/ssh keypair` to be uploaded to remote workspace. And `post_deployment.sh` not nest under folder `default`
1. Upload/download input json to/from sap_library is not in the scope.
1. No shell script reuiqred.

## Solution
1. Create folders in remote exec. Also fix the path in ansible.
1. Remove default value for CIDR and region in tf and add them in deployer.json. Add key landscape in deployer.json
1. Create desired folder structure on deployer.
1. Add `sshkey.pub` and `sshkey` as default key pair path.
1. Remove filename from `backend.tf`.
1. Adjust location and content of `post_deployment.sh`.
1. Remove code that uploads deployer json to sap_library.
1. Remove util/*

## Tests
1. cd ~/sap-hana/deploy/terraform/bootstrap/sap_deployer
1. terraform init && terraform apply -auto-approve -var-file=deployer.json
1. Check on deployer:
```
azureadm@deployer00-vm-9c5ef49d:~$ tree ~
/home/azureadm
└── Azure_SAP_Automated_Deployment
    └── workspaces
        ├── deployer
        ├── local
        │   └── sapdeployer-rg-9c5ef49d
        ├── sap_landscape
        ├── sap_library
        └── sap_system
```
4. And is already logged on with UAI:
```
azureadm@deployer00-vm-9c5ef49d:~$ az account show
{
  "environmentName": "AzureCloud",
  "homeTenantId": "XXX",
  "id": "XXX",
  "isDefault": true,
  "managedByTenants": [],
  "name": "Azure Pipelines for SAP Deployment",
  "state": "Enabled",
  "tenantId": "XXX",
  "user": {
    "assignedIdentityInfo": "MSI",
    "name": "systemAssignedIdentity",
    "type": "servicePrincipal"
  }
}
```
5. Try to upload files to deployer:
```
MININT-LPEJFG4:global-eaus-deployer-infrastructure nancyc$ ./post_deployment.sh 
Start uploading deployer tfstate
terraform.tfstate                                                                                      100%   53KB 204.3KB/s   00:00    
Start uploading deployer json
global-eaus-deployer-infrastructure.json                                                               100%  463     6.1KB/s   00:00    
Start uploading deployer SSH keypair
sshkey                                                                                                 100% 3422    40.9KB/s   00:00    
sshkey.pub                                                                                             100%  774     8.9KB/s   00:00 
```
And on deployer see:
```
azureadm@deployer00-vm-9c5ef49d:~/Azure_SAP_Automated_Deployment/workspaces/local/global-eaus-deployer-infrastructure$ ls -ltr
total 68
-rw-r--r-- 1 azureadm azureadm 54783 Aug 20 23:15 terraform.tfstate
-rw-r--r-- 1 azureadm azureadm   463 Aug 20 23:15 global-eaus-deployer-infrastructure.json
-rw-r--r-- 1 azureadm azureadm  3422 Aug 20 23:15 sshkey
-rw-r--r-- 1 azureadm azureadm   774 Aug 20 23:15 sshkey.pub
```
6. Try ansible playbook on deployer - see no error:

```
~/Azure_SAP_Automated_Deployment$ ansible-playbook -i workspaces/sap_system/global-eaus-sap0-X00/ansible_config_files/hosts sap-hana/deploy/ansible/sap_playbook.yml
...
PLAY RECAP ******************************************************************************************************************************
10.1.1.10                  : ok=44   changed=13   unreachable=0    failed=0    skipped=111  rescued=0    ignored=0   
localhost                  : ok=18   changed=5    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```


## Notes
Close #717
